src/main/java/it/cnr/isti/hpc/wikipedia/article/Link.java

```diff
@@ -100,7 +132,7 @@ public int hashCode() {
 		final int prime = 31;
 		int result = 1;
 		result = prime * result
-				+ ((description == null) ? 0 : description.hashCode());
+				+ ((anchor == null) ? 0 : anchor.hashCode());
```

Comment:
Maybe changing the test to use `.isNull()` or something like that (I never quite remember what is the right way for java)